kong/pdk/tracing.lua

```diff
@@ -575,12 +575,13 @@ local function new_tracer(name, options)
   -- @tparam number sampling_rate the sampling rate to apply for the
   -- probability sampler
   -- @treturn bool sampled value of sampled for this trace
-  function self:get_sampling_decision(parent_should_sample, sampling_rate)
+  function self:get_sampling_decision(parent_should_sample, plugin_sampling_rate)
     local ctx = ngx.ctx
 
     local sampled
     local root_span = ctx.KONG_SPANS and ctx.KONG_SPANS[1]
     local trace_id = tracing_context.get_raw_trace_id(ctx)
+    local sampling_rate = plugin_sampling_rate or kong.configuration.tracing_sampling_rate
```

Comment:
This is a rather indirect way to solve the issue. Should we add a line of comment to explain that we are updating the sample rate after the ID is changed?